scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -64,17 +64,21 @@ interface ScanResultsStorage {
 
         override fun read(id: Identifier) =
                 storage.read(id).also {
-                    ++stats.numReads
-                    if (it.results.isNotEmpty()) {
-                        ++stats.numHits
+                    synchronized(stats) {
+                        ++stats.numReads
+                        if (it.results.isNotEmpty()) {
+                            ++stats.numHits
+                        }
                     }
                 }
 
         override fun read(pkg: Package, scannerDetails: ScannerDetails) =
                 storage.read(pkg, scannerDetails).also {
-                    ++stats.numReads
-                    if (it.results.isNotEmpty()) {
-                        ++stats.numHits
+                    synchronized(stats) {
```

Comment:
nit: instead of putting the synchronized statements into the client code of AccessStatistics 
we could make AccessStatistics a monitor class, e.g. all functions synchronized and add to increment functions there.

Reply:
Or use `AtomicInteger` as suggested here: https://kotlinlang.org/docs/reference/coroutines/shared-mutable-state-and-concurrency.html#thread-safe-data-structures

Reply:
I decided to go for `AtomicInteger` as it seems to fit the specific purpose here best.